PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/sdk/deviceData/managers/UserConsentManager.java

```diff
@@ -80,76 +98,71 @@ public void init(Context context) {
-        updateConsentValue(preferences, GDPR_PREBID_CONSENT);
-        updateConsentValue(preferences, GDPR_PREBID_PURPOSE_CONSENT);
+    private void initConsentValuesAtStart() {
+        for (String consent : GDPR_CONSENTS) {
+            updateConsentValue(sharedPreferences, consent);
+        }
     }
 
     /**
      * Automatically updates consents values in the manager when they are changed.
      */
     private void updateConsentValue(
-        SharedPreferences preferences,
-        @Nullable String key
+            SharedPreferences preferences,
+            @Nullable String key
     ) {
         if (key == null) return;
-
-        switch (key) {
-            case GDPR_PREBID_SUBJECT:
-                // If we couldn't retrieve value from string, it means that user migrated from 1.13.1 where we have boolean value.
-                Object subjectValue = preferences.getAll().get(GDPR_PREBID_SUBJECT);
-                if (subjectValue instanceof String){
-                    gdprPrebidSubject = preferences.getString(GDPR_PREBID_SUBJECT, null);
-                } else if (subjectValue instanceof Boolean){
-                    gdprPrebidSubject = preferences.getBoolean(GDPR_PREBID_SUBJECT, false) ? "1" : "0";
-                    preferences
-                            .edit()
-                            .putString(GDPR_PREBID_SUBJECT,gdprPrebidSubject)
-                            .apply();
-                } else {
-                    gdprPrebidSubject = null;
-                }
-            case GDPR_PREBID_CONSENT:
-                gdprPrebidConsent = preferences.getString(GDPR_PREBID_CONSENT, null);
-            case GDPR_PREBID_PURPOSE_CONSENT:
-                gdprPrebidPurposeConsent = preferences.getString(GDPR_PREBID_PURPOSE_CONSENT, null);
-            case GDPR_1_SUBJECT:
-                gdprSubject = preferences.getString(GDPR_1_SUBJECT, null);
-            case GDPR_1_CONSENT:
-                gdprConsent = preferences.getString(GDPR_1_CONSENT, null);
-            case GDPR_2_CMP_SDK_ID:
-                gdpr2CmpSdkId = preferences.getInt(GDPR_2_CMP_SDK_ID, NOT_ASSIGNED);
-            case GDPR_2_SUBJECT:
-                gdpr2Subject = preferences.getInt(GDPR_2_SUBJECT, NOT_ASSIGNED);
-            case GDPR_2_CONSENT:
-                gdpr2Consent = preferences.getString(GDPR_2_CONSENT, null);
-            case US_PRIVACY_STRING:
-                usPrivacyString = preferences.getString(US_PRIVACY_STRING, null);
-            case GDPR_2_PURPOSE_CONSENT:
-                gdpr2PurposeConsent = preferences.getString(GDPR_2_PURPOSE_CONSENT, null);
-            case COPPA_SUBJECT_CUSTOM_KEY:
-                if (sharedPreferences.contains(COPPA_SUBJECT_CUSTOM_KEY)) {
-                    isSubjectToCoppa = sharedPreferences.getBoolean(COPPA_SUBJECT_CUSTOM_KEY, false);
-                } else {
-                    isSubjectToCoppa = null;
-                }
+        try {
+            switch (key) {
```

Comment:
If you use `switch`, don't forget to add `break` at the end of each case. Now it doesn't work well.